storm-client/src/jvm/org/apache/storm/security/auth/RegexKerberosPrincipalToLocal.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.security.auth;
+
+import java.util.Map;
+import java.security.Principal;
+
+/**
+ * Map a kerberos principal to a local user
+ */
+public class RegexKerberosPrincipalToLocal extends KerberosPrincipalToLocal {
+
+    private String inputRegex;
+	private String replacementString;
+
+	/**
+     * Invoked once immediately after construction
+     * @param storm_conf Storm configuration
+     */
+    public void prepare(@SuppressWarnings("rawtypes") Map storm_conf) {
+        Object object = storm_conf.get("storm.principal.mapper.regex");
+        if (object != null) {
+            inputRegex = object.toString();
+        }
+        object = storm_conf.get("storm.principal.mapper.replacement");
```

Comment:
Could we put all of the configs here inside of Config.java and then tag them with the annotations to make it a string so we don't need to call .toString on the things passed in.